ring/partition_ring.go

```diff
@@ -353,6 +354,68 @@ func (r *PartitionRing) String() string {
 	return fmt.Sprintf("PartitionRing{ownersCount: %d, partitionsCount: %d, partitions: {%s}}", len(r.desc.Owners), len(r.desc.Partitions), buf.String())
 }
 
+func (r *PartitionRing) GetTokenRangesForPartition(partitionID int32) (TokenRanges, error) {
```

Comment:
[nit] Please add a short comment to explain this function. In particular, explain the partition state is not took in account. If you want to get token ranges only looking at Active partitions then you should grab the subring via `ShuffleShard()` first.

Reply:
Done in [`7f9f1bf` (#488)](https://github.com/grafana/dskit/pull/488/commits/7f9f1bf7b6b2b0ad4599d23690fc88ad66db4ec0).